tagstudio/src/qt/ts_qt.py

```diff
@@ -393,11 +395,27 @@ def start(self):
 				lib = self.settings.value("last_library")
 			self.splash.showMessage(f'Opening Library "{lib}"...', int(Qt.AlignmentFlag.AlignBottom|Qt.AlignmentFlag.AlignHCenter), QColor('#9782ff'))
 			self.open_library(lib)
+		
+		def refresh_library():
+			self.lib.refresh_dir()
+			self.preview_panel.update_widgets()
+			self.filter_items()
+
+		class IsDirModifiedHandler(FileSystemEventHandler):
+			def on_any_event(self, event):
+				if not event.is_directory:
+					refresh_library()
+		
+		observer = Observer()
+		observer.schedule(IsDirModifiedHandler(), lib, recursive=True)
+		observer.start()
```

Comment:
I think this should be moved out of the start method.
The class should be defined somewhere else and should not be a part of the QtDriver class. I think refresh_library should be a class method of QtDriver and not be defined in start().